src/core/Akka/Util/Internal/AtomicState.cs

```diff
@@ -113,6 +113,38 @@ public async Task<T> CallThrough<T>(Func<Task<T>> task)
             }
             return result;
         }
+        
+        public async Task<T> CallThrough<T,TState>(TState state, Func<TState,Task<T>> task)
```

Comment:
Safe to use `ValueTask` here? I don't know of any instances where multiple `await`ers await on the same `CircuitBreaker` at the same time off the top of my head. Maybe @ismaelhamed does?

Reply:
You can call `WithSyncCircuitBreaker` multiple times without waiting for the previous ones to finish, if that was the question. I'm thinking CircuitBreaker + PipeTo.

Reply:
The `ValueTask` constraint I'm thinking of here is that you can't have multiple awaiters on the same `ValueTask` - if `WithCircuitBreaker` is called multiple times those are all different tasks right? Or is that method returning the same one each time possibly?

Reply:
@Aaronontheweb I think there are spots right now where we `return` the task all the way up the callchain to `WithCircuitBreaker`, if so it would propagate up there into public API space.

If we want to further reduce allocations we may want to consider adapting something like PooledAwait (which uses special async method builders to avoid some of the state machine costs, also they have a fancy thing that replaces TCS, lowering allocation on most platforms with safe fallback if not.)

Reply:
Ok, so `ValueTask` is out then under those conditions. Got it.